neo/Network/RPC/RpcServer.cs

```diff
@@ -297,7 +297,21 @@ private JObject Process(string method, JArray _params)
                         return json;
                     }
                 case "getrawmempool":
-                    return new JArray(Blockchain.Singleton.GetMemoryPool().Select(p => (JObject)p.Hash.ToString()));
+                    {
+                        bool shouldGetUnverified = _params.Count >= 1 && _params[0].AsBooleanOrDefault(false);
+                        if (!shouldGetUnverified)
+                            return new JArray(Blockchain.Singleton.GetMemoryPoolVerified().Select(p => (JObject)p.Hash.ToString()));
+
+                        JObject json = new JObject();
+                        Blockchain.Singleton.GetMemoryPoolSeparateVerfiedAndUnverified(
+                            out IEnumerable<Transaction> verifiedTransactions,
+                            out IEnumerable<Transaction> unverifiedTransactions);
+                        json["height"] = Blockchain.Singleton.Height;
```

Comment:
Although `Blockchain.Singleton.Height` is called immediately after `GetMemoryPoolSeparateVerfiedAndUnverified()`, consistency is not guaranteed. It is recommended to get the block height first, then call `GetMemoryPoolSeparateVerfiedAndUnverified()`.

Reply:
I’ll switch to reading the height before as you suggest.

Reply:
Reading it before still doesn’t guarantee consistency unfortunately. Perhaps it Would better if the MemoryPool also kept the height for which it has last verified a transaction that it also protects with the write lock. Then it can also output that from `GetMemoryPoolSeparateVerifiedAndUnverified`.

Reply:
Transactions that are verified at lower height may not be verifiable at higher height; however, transactions that are verified at higher height are safe for lower height states.

Reply:
It makes sense. I agree it is fine to leave it this way.